operator-common/src/test/java/io/strimzi/operator/common/DefaultAdminClientProviderTest.java

```diff
@@ -97,10 +103,35 @@ public void testMTlsWithPublicCAConnection() {
 
     @Test
     public void testNullConfig() {
-        InvalidConfigurationException ex = assertThrows(InvalidConfigurationException.class, () -> DefaultAdminClientProvider.adminClientConfiguration(null, mockPemAuthIdentity(), null));
+        DefaultAdminClientProvider defaultAdminClientProvider = new DefaultAdminClientProvider();
+        InvalidConfigurationException ex = assertThrows(InvalidConfigurationException.class, () -> defaultAdminClientProvider.adminClientConfiguration(null, mockPemAuthIdentity(), null));
         assertThat(ex.getMessage(), is("The config parameter should not be null"));
     }
 
+    @Test
+    public void tesCreateControllerAdminClient() {
+        DefaultAdminClientProvider defaultAdminClientProvider = mock(DefaultAdminClientProvider.class);
+        when(defaultAdminClientProvider.adminClientConfiguration(any(), any(), any())).thenAnswer(invocation -> {
+            Properties config = invocation.getArgument(2);
+            assertThat(config.size(), is(1));
+            assertThat(config.getProperty(AdminClientConfig.BOOTSTRAP_CONTROLLERS_CONFIG), is("my-kafka-controller:9090"));
+            return config;
+        });
+        defaultAdminClientProvider.createControllerAdminClient("my-kafka-controller:9090", null, null, new Properties());
+    }
+
+    @Test
+    public void testCreateBrokerAdminClient() {
+        DefaultAdminClientProvider defaultAdminClientProvider = mock(DefaultAdminClientProvider.class);
+        when(defaultAdminClientProvider.adminClientConfiguration(any(), any(), any())).thenAnswer(invocation -> {
+            Properties config = invocation.getArgument(2);
+            assertThat(config.size(), is(1));
+            assertThat(config.getProperty(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG), is("my-kafka:9092"));
+            return config;
+        });
+        defaultAdminClientProvider.createAdminClient("my-kafka:9092", null, null, new Properties());
+    }
```

Comment:
I think this is an imaginative ay to test it which I kind of like. But you should probably have some kind of CountDownLatch there or something to actually make sure the asserts were called?

Reply:
You are right. I think I need to use slightly different approach to make sure the asserts are called. I've updated the last commit I pushed.

Reply:
Thanks.